package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,302 @@
+        # used to do that.
+        nframessize = filesize - self.header_size - self.firstframesize
+        return nframessize / self.framesize + 1
+
+    @property
+    def periodic(self):
+          return bool((self.charmm & DCD_IS_CHARMM) and
+                      (self.charmm & DCD_HAS_EXTRA_BLOCK))
+
+
+    def read(self):
+        if self.reached_eof:
+            raise RuntimeError('Reached last frame in TRR, seek to 0')
+        if not self.is_open:
+            raise RuntimeError("No file open")
+        if self.mode != 'r':
+            raise RuntimeError('File opened in mode: {}. Reading only allow '
+                               'in mode "r"'.format('self.mode'))
+
+        cdef np.ndarray xyz = np.empty((self.n_atoms, 3), dtype=DTYPE,
+                                       order='F')
+        cdef np.ndarray unitcell = np.empty(6, dtype=DTYPE)
+
+        cdef DTYPE_t[::1] x = xyz[:, 0]
+        cdef DTYPE_t[::1] y = xyz[:, 1]
+        cdef DTYPE_t[::1] z = xyz[:, 2]
+
+        first_frame = self.current_frame == 0
+
+        ok = read_dcdstep(self.fp, self.n_atoms, <DTYPE_t*> &x[0],
+                          <DTYPE_t*> &y[0], <DTYPE_t*> &z[0],
+                          <DTYPE_t*> unitcell.data, self.nfixed, first_frame,
+                          self.freeind, self.fixedcoords,
+                          self.reverse_endian, self.charmm)
+        if ok != 0 and ok != -4:
+            raise IOError("Reading DCD header failed: {}".format(DCD_ERRORS[ok]))
+
+        # we couldn't read any more frames.
+        if ok == -4:
+            self.reached_eof = True
+            raise StopIteration
+
+        self.current_frame += 1
+
+        return DCDFrame(xyz, unitcell)
+
+    def seek(self, frame):
+        """Seek to Frame.
+
+        Please note that this function will generate internal file offsets if
+        they haven't been set before. For large file this means the first seek
+        can be very slow. Later seeks will be very fast.
+
+        Parameters
+        ----------
+        frame : int
+            seek the file to given frame
+
+        Raises
+        ------
+        IOError
+            If you seek for more frames than are available or if the
+            seek fails (the low-level system error is reported).
+        """
+        if frame >= self.n_frames:
+            raise IOError('Trying to seek over max number of frames')
+        self.reached_eof = False
+
+        cdef fio_size_t offset
+        if frame == 0:
+            offset = self.header_size
+        else:
+            offset = self.header_size + self.firstframesize + self.framesize * (frame - 1)
+
+        ok = fio_fseek(self.fp, offset, _whence_vals["FIO_SEEK_SET"])
+        if ok != 0:
+            raise IOError("DCD seek failed with system errno={}".format(ok))
+        self.current_frame = frame
+
+    def tell(self):
+        """Get current frame"""
+        return self.current_frame
```

Comment:
Looks like you already defined the `tell()` method above? Sorry -- I should have noticed that, but I just realized that my test was passing (once you told me about `tell`) before I fetched your update to this branch.